src/MediatR/Pipeline/RequestExceptionActionProcessorBehavior.cs

```diff
@@ -71,7 +72,7 @@ private IEnumerable<(Type ExceptionType, object Action)> GetActionsForException(
         var exceptionActionInterfaceType = typeof(IRequestExceptionAction<,>).MakeGenericType(typeof(TRequest), exceptionType);
         var enumerableExceptionActionInterfaceType = typeof(IEnumerable<>).MakeGenericType(exceptionActionInterfaceType);
 
-        var actionsForException = (IEnumerable<object>)_serviceProvider.GetService(enumerableExceptionActionInterfaceType);
+        var actionsForException = (IEnumerable<object>)_serviceProvider.GetRequiredService(enumerableExceptionActionInterfaceType);
```

Comment:
What are the implications of changing this to GetRequiredService?

Reply:
[GetRequiredService](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getrequiredservice?view=dotnet-plat-ext-6.0#microsoft-extensions-dependencyinjection-serviceproviderserviceextensions-getrequiredservice(system-iserviceprovider-system-type)) throws a `InvalidOperationException` when the requested service type was **not** registered.

[GetService](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getservice?view=dotnet-plat-ext-6.0#returns) returns `null` when the requested service was not registered. This behavior is not framework-specific. In the current state of the master branch, this could _eventually_ happen. The following cast to (`IEnumerable<object>`) would have thrown a null-ref exception.

It's either we use `GetRequiredService` or we suppress the nullable warning with `!` to say that we are 100% sure that there will be a registration for the requested type.

Reply:
I'm going to add a unit test for this scenario. `GetService` should *never* return null for enumerables, it would be an empty list. But I don't know what the behavior would be for `GetRequiredService` for enumerables, would it throw if there aren't any registrations? I wouldn't think so, but work verifying.

Reply:
Oh it's a little bit late but there is also a built-in [IServiceProvider.GetServices(serviceType)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getservices?view=dotnet-plat-ext-6.0) which returns an `IEnumerable<T>`, so basically it returns an empty list if there's no registration for `T`.